File: Swift/shortest-unsorted-continuous-subarray.swift
class Solution {
    func findUnsortedSubarray(_ nums: [Int]) -> Int {
        guard nums.count > 1 else {
            return 0
        }
        let sortedNums = nums.sorted()
        var p = -1, q = -1

        for (index, value) in zip(nums, sortedNums).enumerated() {
            if value.0 != value.1 {
                p == -1 ? (p = index) : (q = index + 1)
            }
        }
        /*
        for index in 0..<nums.count {
            if nums[index] != sortedNums[index] {
                p == -1 ? (p = index) : (q = index + 1)
            }
        }
        */

        return q - p
    }
}

File: Python/shortest-unsorted-continuous-subarray.py

class Solution:
    def findUnsortedSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) < 2:
            return 0

        sortedNums = sorted(nums)
        p = -1
        q = -1
        
        for i in range(len(nums)):
            if nums[i] != sortedNums[i]:
                if p == -1:
                    p = i
                else:
                    q = i + 1
        return q - p
